src/lib/units/setTime.js

```diff
@@ -0,0 +1,55 @@
+import { getParseRegexForToken, isoTimes }   from '../parse/regex';
+import { prepareConfig } from '../create/from-anything';
+import { copyConfig } from '../moment/constructor';
+import { parseFromStringAndFormat } from '../create/from-string-and-format';
+import { HOUR, MINUTE, SECOND, MILLISECOND } from '../units/constants';
+import { isValid } from '../create/valid';
+import getParsingFlags from '../create/parsing-flags';
+
+
+export function setTime(input, format) {
+
+    if (!this.isValid())
+    {
+        return this;
+    }
+
+    if (typeof input === 'number')
+    {
+        return this.startOf('day').add(input, 'm');
+    }
+
+    if (!format)
+    {
+        format = findIsoFormat(input);
+    }
+
+    if (!format) {
+        this.invalidFormat = true;
+    }
+
+    this._f = format;
+    this._i = input;
+    this._isValid = null;
+
+    parseFromStringAndFormat(this);
```

Comment:
@mj1856 this isn't a bug, because I don't _think_ it affects functionality, but the copyConfig function in src\lib\moment\constructor.js doesn't work right.
Consider the following code, keeping in mind that the isDSTShifted function calls the copyConfig function, and then calls the parser:

    var a = moment('2015-02-03', 'YYYY-MM-DD a');
    var arr = [ 'a' ];
    assert.deepEqual(a._pf.unusedTokens, arr);
    a.isDSTShifted();
    assert.deepEqual(a._pf.unusedTokens, arr);

The first test passes, the second test fails as the value of a._pf.unusedTokens is now [ 'a', 'a' ]. This is because _pf is a reference value, not a primitive. 
To have setTime properly work on a copy, I have to change how the copyConfig function works. As it's used in a couple other places, including the constructor, I guess I just want to document that this behavior is changing, on the off chance it was desired for some reason.